tests/trilinos/nox_solver_01.cc

```diff
@@ -0,0 +1,99 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Check TrilinosWrappers::NOXSolver.
```

Comment:
I think it would be nice if you could adjust the various kinsol tests into NOX tests as well. You chose the interfaces pretty similar, I believe, so this should be too much work, but it would cover a number of interesting cases.

Reply:
Separately, can you document what it is this test is actually solving? I believe it's finding the root x^2==0, but I'm not entirely sure.

Reply:
> I think it would be nice if you could adjust the various kinsol tests into NOX tests as well. You chose the interfaces pretty similar, I believe, so this should be too much work, but it would cover a number of interesting cases.

Would be nice. But I won't have time to this in the next weeks. Nevertheless, I would like to get the wrappers in the library so that peoples can use them and we can extend the functionality centrally.